app/controllers/dfe_sign_in_controller.rb

```diff
@@ -12,14 +16,19 @@ def callback
 private
 
   def default_authenticated_path
-    if authorized_for_support_interface?
+    if get_support_user
       support_interface_path
     else
       provider_interface_path
     end
   end
 
-  def authorized_for_support_interface?
-    SupportUser.load_from_session(session)
+  def get_support_user
+    if @already_tried # using separate var as guard as lookup can return nil, which is then cached
+      @user
+    else
+      @already_tried = true
+      @user = SupportUser.load_from_session(session)
+    end
```

Comment:
Suggestion:

```rb
@user == nil ? (SupportUser.load_from_session(session) || false) : @user
```

Reply:
Nice, thanks, was looking for a more concise version.

Reply:
Went with:

```ruby
@user != nil ? @user : @user = (SupportUser.load_from_session(session) || false)
```